publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -382,6 +386,28 @@ public List<LogEntry> getLogEntriesForResource(Resource resource) {
                     .toList();
     }
 
+    public void persistFile(FileEntry fileEntry) {
+        var dao = fileEntry.toDao().createInsertionTransactionRequest();
+        sendTransactionWriteRequest(dao);
+    }
+
+    public Optional<FileEntry> fetchFile(FileEntry fileEntry) {
+        var primaryKey = fileEntry.toDao().primaryKey();
+        var result = client.getItem(new GetItemRequest().withTableName(tableName).withKey(primaryKey));
+        return Optional.ofNullable(result.getItem())
+                   .map(FileDao::fromDynamoFormat)
+                   .map(FileEntry::fromDao);
+    }
+
+    public void deleteFile(FileEntry fileEntry) {
+        var primaryKey = fileEntry.toDao().primaryKey();
+        client.deleteItem(new DeleteItemRequest().withTableName(tableName).withKey(primaryKey));
+    }
+
+    public void updateFile(FileEntry fileEntry) {
+        fileEntry.toDao().updateExistingEntry(client);
```

Comment:
Should maybe make sure if someone tries to change file reference/id that it is stopped

Reply:
no worries, already done I see :-) identifiersDoesNotMatch check